feat: 게시글 사진 삽입 기능 추가

yscarlet opened the pull request:
### 🐣Title
게시글 사진 삽입 기능 추가



### 🐣Part
BE


### 🐣Key Changes
게시글 사진 삽입 기능 추가를 위한 FileUploadController, FileStorageService, FileStorageProperties 파일 추가
데이터베이스 연결 확인을 위한 test 코드 작성
dto access level 수정

### 🐣To Reviewer
- 게시글에서의 사진 삽입을 보편적인 블로그 식으로 구현하기 위하여 content는 html tag를 가진채로 저장됩니다. user가 사진을 업로드 할 경우 image는 서버로 저장되며 서버에 저장된 이미지를 다시 front가 불러와 화면에 나타내는 식으로 진행됩니다. (추가 test 및 검증 필요)
- user를 생성할 시 user의 nickname과 introduce를 사용하여 명시적으로 생성하는 방식이 막혀있기 때문에 test는 local의 database에 test에 필요한 user와 category의 index를 미리 생성시킨 뒤 진행했습니다.

### 🐣Next
게시글 사진 삽입 기능 보완 및 추가 검증
postman을 사용하여 api 추가 검증
동시성 문제 처리 예정

yscarlet committed:
feat:Post content 사진 추가 기능 추가

기존의 텍스트 형식만 지원하던 content에서 사진 추가 기능을 추가함

yscarlet committed:
test: post 데이터베이스 관련 test

데이터베이스에서 삽입, 삭제, 수정이 제대로 이루어지는지 확인

yunuo46 commented on src/test/java/com/gaplog/server/PostTest.java:
```diff
@@ -0,0 +1,38 @@
+        postService.PostLikeUpdate(1L, postLikeUpdateRequest);
+        SelectedPostResponse selectedPostResponse = postService.getSelectedPostInfo(1L);
+
+        System.out.println(selectedPostResponse.getId());
+        System.out.println(selectedPostResponse.getTitle());
+        System.out.println(selectedPostResponse.getContent());
+        System.out.println(selectedPostResponse.getLikeCount());
+        System.out.println(selectedPostResponse.getJinjiCount());
+        System.out.println(selectedPostResponse.getCreatedAt());
+        System.out.println(selectedPostResponse.getUpdatedAt());
```
junut의 assert메소드를 사용해주세요

yunuo46 commented on src/test/java/com/gaplog/server/PostTest.java:
테스트 시에는 h2 인메모리 DB를 사용하시는 걸 고려해보세요

yunuo46 replied:
테스트 코드를 시도해보신 점 아주 좋습니다!! 좀 더 커버리지를 높여보세요 :)

yunuo46 commented on src/test/java/com/gaplog/server/PostTest.java:
```diff
@@ -0,0 +1,38 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
```
@Transactional 혹은 @AfterEach를 사용하여 롤백해주세요

yscarlet committed:
feat: 진지버튼 비공개 기능 추가 및 soft delete 기능 추가

진지버튼 4개 획득시 게시물 비공개로 전환 기능 추가
게시물 삭제시 soft delete로 삭제되도록 수정

yscarlet committed:
fix: isDeleted 데이터베이스 반영 오류 해결

postService의 delete 기능 수행시 db에 반영되지 않았던 오류 해결

yscarlet committed:
feat: 카테고리 수정기능 추가

updatePost를 통해 카테고리도 수정할 수 있도록 수정
db 적용 확인 완

yscarlet committed:
refactor: jinji 변수명 변경 및 파일명 변경

jinji->seriousness

yscarlet committed:
fix: api request 수정(api db 연결 확인)

postApi- 특정게시물 조회 - request body 삭제
postRequest - user 자료형:User-> Long으로 변경

yscarlet committed:
fix: Seriousness 오류 해결

4개의 진지버튼을 얻어 게시글이 비공개 처리 된 후 다시 진지버튼을 누르면 3으로 감소하는 오류 해결

yscarlet committed:
refactor: 어노테이션을 통한 softdelete구현

-SQLDelete를 통한 soft delete 구현(기존: updateIsDeleted를 통한 자체 구현)
-Where 어노테이션 추가를 통해 게시글 조회시 삭제된 게시글은 걸러지도록 수정

yscarlet committed:
fix: 낙관적 락을 통한 동시성 이슈 해결

좋아요, 진지버튼을 누르는 과정에서 발생할 수 있는 충돌문제를 낙관적 락을 걸어 해결

yscarlet committed:
feat: 메인 게시글 조회

메인 페이지에서 최신 게시물 20개를 업로드 순으로 볼 수 있는 api추가

yunuo46 committed:
Merge branch 'main' into main
